src/components/Header/NavBar/Authorization/Authorization.js

```diff
@@ -1,23 +1,15 @@
-import React, { Component } from 'react';
+import { Component } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { NavLink, withRouter } from 'react-router-dom';
+import { withRouter } from 'react-router-dom';
 
 import { logOut } from '../../../../actions';
 
 export class Authorization extends Component {
-    rendererNav = () => {
-      const { handleLogOut, auth } = this.props;
-      if (auth) return <NavLink to="/login" onClick={handleLogOut}>Log out</NavLink>;
-
-      return <NavLink to="/login">Login</NavLink>;
-    }
-
-    render() {
-      return (
-        this.rendererNav()
-      );
-    }
+  render() {
+    const { children, auth, handleLogOut } = this.props;
+    return children(auth, handleLogOut);
```

Comment:
Good that you managed to figure out the pattern of passing children to the component, but it's not what I meant.

https://trello.com/c/zbrPnunH/39-extract-conditional-rendering-depending-on-authorization-to-separate-component-using-render-props